solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/EuristicBoostedModel.java

```diff
@@ -0,0 +1,150 @@
+package org.apache.solr.ltr.ranking;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.Explanation;
+import org.apache.solr.ltr.feature.LTRScoringAlgorithm;
+import org.apache.solr.ltr.util.ModelException;
+import org.apache.solr.ltr.util.NamedParams;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+public abstract class EuristicBoostedModel extends LTRScoringAlgorithm {
+  public static final String EURISTIC_BOOST_PARAM = "boost";
+  public static final String ADDITIVE_BOOST = "additive";
+  public static final String MULTIPLICATIVE_BOOST = "multiplicative";
+
+
+  protected EuristicFeatureBoost euristicFeatureBoost;
+  protected LTRScoringAlgorithm LTRModel;
+
+  protected float internalModelScore;
+  protected float boostFeatureValue;
```

Comment:
I don't think you can keep these as member variables to carry over between the explain() and score().  1 LTRScoringAlgorithm instance is shared between many queries using the exact same model.  It might be the case that Query1 is executing the explain, and in the middle Query2 using the same model has score() called, overwriting these variables, making your Query1 explain wrong.  Because of this ScoringAlgorithms should not really do anything to its state in score/explain calls.